File: python/CaptureDevice.py
from Item import Item

# buat kelas capture device
class CaptureDevice(Item):
    def __init__(self, id_Item: str = "", nama_Item: str = "", harga_item: float = 0.0, deskripsi_Item: str = "",
                 levelCap: str = "", typeDevice: str = "", isUsable: int = 0):
        super().__init__(id_Item, nama_Item, harga_item, deskripsi_Item)
        self.__levelCap_CaptureDevice = str(levelCap)  # buat level cap device
        self.__type_CaptureDevice = str(typeDevice)    # buat tipe device
        self.__isUsable_CaptureDevice = int(isUsable)  # buat status usable (1: true, 0: false)

    # buat setter level cap
    def setLevelCap_CaptureDevice(self, levelCap):
        self.__levelCap_CaptureDevice = str(levelCap)

    def setLevelCap(self, levelCap):
        self.__levelCap_CaptureDevice = str(levelCap)

    # buat setter type device
    def setType_CaptureDevice(self, typeDevice):
        self.__type_CaptureDevice = str(typeDevice)

    def setType(self, typeDevice):
        self.__type_CaptureDevice = str(typeDevice)

    # buat setter is usable
    def setIsUsable_CaptureDevice(self, isUsable):
        self.__isUsable_CaptureDevice = int(isUsable)

    def setIsUsable(self, isUsable):
        self.__isUsable_CaptureDevice = int(isUsable)

    # buat getter level cap
    def getLevelCap_CaptureDevice(self):
        return self.__levelCap_CaptureDevice

    def getLevelCap(self):
        return self.__levelCap_CaptureDevice

    # buat getter type device
    def getType_CaptureDevice(self):
        return self.__type_CaptureDevice

    def getType(self):
        return self.__type_CaptureDevice

    # buat getter is usable
    def getIsUsable_CaptureDevice(self):
        return self.__isUsable_CaptureDevice

    def getIsUsable(self):
        return self.__isUsable_CaptureDevice

    # buat cek status usable
    def getKeteranganIsUsable(self):
        if self.__isUsable_CaptureDevice == 1:
            return "Bisa digunakan"
        else:
            return "Tidak bisa digunakan"

    # buat cek is usable (boolean)
    def cekIsUsable(self):
        return self.__isUsable_CaptureDevice == 1

    # buat show data capture device
    def showDataCaptureDevice(self):
        print(f"Level Cap Device   : {self.__levelCap_CaptureDevice}")
        print(f"Type Device        : {self.__type_CaptureDevice}")
        usable_str = "1 (True / Bisa digunakan)" if self.__isUsable_CaptureDevice == 1 else "0 (False / Tidak bisa digunakan)"
        print(f"Is Usable          : {usable_str}")

File: php/CaptureDevice.php
<?php
require_once 'Item.php';

// buat kelas capture device
class CaptureDevice extends Item {
    private string $levelCap_CaptureDevice; // buat level cap device
    private string $type_CaptureDevice;     // buat tipe device
    private int $isUsable_CaptureDevice;    // buat status usable device (1: true, 0: false)

    // constructor default & parametrik
    public function __construct(
        string $id = "",
        string $nama = "",
        float $harga = 0.0,
        string $deskripsi = "",
        string $gambar = "",
        string $levelCap = "",
        string $type = "",
        int $isUsable = 0
    ) {
        parent::__construct($id, $nama, $harga, $deskripsi, $gambar);
        $this->levelCap_CaptureDevice = $levelCap;
        $this->type_CaptureDevice = $type;
        $this->isUsable_CaptureDevice = $isUsable;
    }

    // buat setter level cap
    public function setLevelCap_CaptureDevice(string $levelCap): void {
        $this->levelCap_CaptureDevice = $levelCap;
    }

    public function setLevelCap(string $levelCap): void {
        $this->levelCap_CaptureDevice = $levelCap;
    }

    // buat setter type device
    public function setType_CaptureDevice(string $type): void {
        $this->type_CaptureDevice = $type;
    }

    public function setType(string $type): void {
        $this->type_CaptureDevice = $type;
    }

    // buat setter is usable
    public function setIsUsable_CaptureDevice(int $isUsable): void {
        $this->isUsable_CaptureDevice = $isUsable;
    }

    public function setIsUsable(int $isUsable): void {
        $this->isUsable_CaptureDevice = $isUsable;
    }

    // buat getter level cap
    public function getLevelCap_CaptureDevice(): string {
        return $this->levelCap_CaptureDevice;
    }

    public function getLevelCap(): string {
        return $this->levelCap_CaptureDevice;
    }

    // buat getter type device
    public function getType_CaptureDevice(): string {
        return $this->type_CaptureDevice;
    }

    public function getType(): string {
        return $this->type_CaptureDevice;
    }

    // buat getter is usable
    public function getIsUsable_CaptureDevice(): int {
        return $this->isUsable_CaptureDevice;
    }

    public function getIsUsable(): int {
        return $this->isUsable_CaptureDevice;
    }

    // buat cek status usable (apakah bisa dipakai berkali-kali)
    public function getKeteranganIsUsable(): string {
        return ($this->isUsable_CaptureDevice === 1) ? "Bisa dipakai berkali-kali" : "Sekali pakai";
    }

    // buat cek usable (boolean)
    public function cekIsUsable(): bool {
        return $this->isUsable_CaptureDevice === 1;
    }

    // buat show data capture device
    public function showDataCaptureDevice(): void {
        echo "Level Cap Device   : " . $this->levelCap_CaptureDevice . "<br>";
        echo "Type Device        : " . $this->type_CaptureDevice . "<br>";
        $usableStr = ($this->isUsable_CaptureDevice === 1) ? "1 (Bisa dipakai berkali-kali)" : "0 (Sekali pakai)";
        echo "Bisa Pakai Berkali-kali : " . $usableStr . "<br>";
    }
}
?>
